lib/srv/desktop/discovery.go

```diff
@@ -113,7 +112,7 @@ func (s *WindowsService) ldapSearchFilter() string {
 
 // getDesktopsFromLDAP discovers Windows hosts via LDAP
 func (s *WindowsService) getDesktopsFromLDAP() types.ResourcesWithLabels {
-	if atomic.LoadInt32(&s.ldapInitialized) == 0 {
+	if !s.ldapReady() {
```

Comment:
I think this check here has the potential to cause us issues:

Consider the case that in one reconcile loop,
`readWithFilter` (line 123 below this) fails and we call `s.initializeLDAP` again (line 130). If that `initializeLDAP` fails, then inside it we'll be setting `s.ldapInitialized = false`. So on the next reconcile loop, this check will be triggered and we'll return `nil`, but won't try to `initializeLDAP` again, and will do so for every subsequent reconcile loop.